common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -274,92 +275,145 @@ func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutions(
 }
 
 func (v *pinotVisibilityTripleManager) ListClosedWorkflowExecutionsByStatus(
 	ctx context.Context,
 	request *ListClosedWorkflowExecutionsByStatusRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager, manager2 := v.chooseVisibilityManagerForRead(request.Domain)
+	if manager2 != nil {
+		return comparePinotESListClosedByStatusResponse(ctx, manager, manager2, request, v.logger)
+	}
 	return manager.ListClosedWorkflowExecutionsByStatus(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) GetClosedWorkflowExecution(
 	ctx context.Context,
 	request *GetClosedWorkflowExecutionRequest,
 ) (*GetClosedWorkflowExecutionResponse, error) {
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager, manager2 := v.chooseVisibilityManagerForRead(request.Domain)
+	if manager2 != nil {
+		return comparePinotESGetClosedByStatusResponse(ctx, manager, manager2, request, v.logger)
+	}
 	return manager.GetClosedWorkflowExecution(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsByQueryRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager, manager2 := v.chooseVisibilityManagerForRead(request.Domain)
+	if manager2 != nil {
+		return comparePinotESListByQueryResponse(ctx, manager, manager2, request, v.logger)
+	}
 	return manager.ListWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ScanWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsByQueryRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager, manager2 := v.chooseVisibilityManagerForRead(request.Domain)
+	if manager2 != nil {
+		return comparePinotESScanResponse(ctx, manager, manager2, request, v.logger)
+	}
 	return manager.ScanWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) CountWorkflowExecutions(
 	ctx context.Context,
 	request *CountWorkflowExecutionsRequest,
 ) (*CountWorkflowExecutionsResponse, error) {
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager, manager2 := v.chooseVisibilityManagerForRead(request.Domain)
+	if manager2 != nil {
+		return comparePinotESCountResponse(ctx, manager, manager2, request, v.logger)
+	}
 	return manager.CountWorkflowExecutions(ctx, request)
 }
 
-func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForRead(domain string) VisibilityManager {
+func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForRead(domain string) (VisibilityManager, VisibilityManager) {
 	var visibilityMgr VisibilityManager
+	var visibilityMgr2 VisibilityManager
+	if v.readModeIsBoth(domain) {
+		if v.esVisibilityManager != nil {
+			visibilityMgr = v.esVisibilityManager
+		} else {
+			visibilityMgr = v.pinotVisibilityManager
```

Comment:
We also need to make sure pinotVisibilityManager != nil before we can return

Reply:
done